lib/dao.js

```diff
@@ -415,7 +415,9 @@ DataAccessObject.updateOrCreate = DataAccessObject.upsert = function upsert(data
         var update = data;
         var inst = data;
         if(!(data instanceof Model)) {
+          console.log('not instance', data);
           inst = new Model(data);
```

Comment:
Maybe the constructor should have an option to skip default value generation in this case.